ios/Classes/NativeBridge/Handlers/MobileAuthHandler.swift

```diff
@@ -2,132 +2,128 @@ import Foundation
             return
         }
-        
-        challenge.sender.respond(withData: otp, challenge: challenge)
+
+        callback(true)
+        mobileAuthCallback = nil
+        completion(.success)
     }
-    
-    func handleQrOTPMobileAuth(_ otp: String , customRegistrationChallenge: ONGCustomRegistrationChallenge?, _ completion: @escaping (Any?, SdkError?) -> Void) {
-        mobileAuthCompletion = completion
-        guard ONGUserClient.sharedInstance().canHandleOTPMobileAuthRequest(otp) else {
-            completion(false, SdkError(.cantHandleOTP))
+
+    func denyMobileAuthRequest(completion: @escaping (Result<Void, FlutterError>) -> Void) {
+        Logger.log("denyMobileAuthRequest", sender: self)
+        guard let callback = mobileAuthCallback else {
+            completion(.failure(FlutterError(SdkError(.otpAuthenticationNotInProgress))))
             return
         }
-        ONGUserClient.sharedInstance().handleOTPMobileAuthRequest(otp, delegate: self)
+
+        callback(false)
+        mobileAuthCallback = nil
+        completion(.success)
+    }
+
+    func setMobileAuthCallback(didReceiveConfirmationChallenge confirmation: @escaping (Bool) -> Void) {
+        mobileAuthCallback = confirmation
+    }
+
+    func finishMobileAuthenticationFlow() {
+        flowStarted = false
     }
 }
 
-//MARK: - ONGMobileAuthRequestDelegate
-extension MobileAuthHandler: ONGMobileAuthRequestDelegate {
-    func userClient(_: ONGUserClient, didReceiveConfirmationChallenge confirmation: @escaping (Bool) -> Void, for request: ONGMobileAuthRequest) {
-        authenticatorType = .confirmation
-        self.confirmation = confirmation
-        mobileAuthCompletion?(request.message, nil)
-        //FIXME: This message can clearly be nullable, we should support this.
-        SwiftOneginiPlugin.flutterApi?.n2fOpenAuthOtp(message: request.message ?? "") {}
+//MARK: - MobileAuthRequestDelegate
+class MobileAuthDelegate: MobileAuthRequestDelegate {
+    private var handleMobileAuthCompletion: (Result<Void, FlutterError>) -> Void
+
+    init(handleMobileAuthCompletion: @escaping (Result<Void, FlutterError>) -> Void) {
+        self.handleMobileAuthCompletion = handleMobileAuthCompletion
+    }
+
+    func userClient(_ userClient: UserClient, didReceiveConfirmation confirmation: @escaping (Bool) -> Void, for request: MobileAuthRequest) {
+        BridgeConnector.shared?.toMobileAuthHandler.setMobileAuthCallback(didReceiveConfirmationChallenge: confirmation)
+        SwiftOneginiPlugin.flutterApi?.n2fOpenAuthOtp(message: request.message) {}
     }
 
-    func userClient(_: ONGUserClient, didReceive challenge: ONGPinChallenge, for request: ONGMobileAuthRequest) {
-       //@todo will need this for PUSH
+    func userClient(_ userClient: UserClient, didReceivePinChallenge challenge: PinChallenge, for request: MobileAuthRequest) {
+        //@todo will need this for PUSH
```

Comment:
why not `// TODO:` ?

Reply:
changed; it was already like this before the rework but you were right